src/bots/unslashed-kleros-connector/execute-timeouts.js

```diff
@@ -0,0 +1,81 @@
+// Eventless bot.
+// The logic of this bot does not query events in order to determine when to call acceptClaim and refuseClaim on the KlerosConnector contract.
+const delay = require('delay')
+
+const ClaimStatus = {
+  None: 0,
+  Created: 1,
+  Disputed: 2,
+  Readjustable: 3,
+  Resolved: 4
+}
+
+module.exports = async (web3, batchedSend, klerosConnector) => {
+  // Keep track of the claims so that we don't waste unnecessary resources on them.
+  const claims = {}
+  const queryFrequency = 60 // minutes
+
+  while (true) {
+    const contractState = {
+      nbClaims: await klerosConnector.methods.nbClaims().call(),
+      challengeTimeout: await klerosConnector.methods.challengeTimeout().call(),
+      readjustmentTimeout: await klerosConnector.methods.readjustmentTimeout().call()
+    }
+    const minQueryInterval =
+      contractState.challengeTimeout < contractState.readjustmentTimeout
+        ? contractState.challengeTimeout
+        : contractState.readjustmentTimeout
+    const transactionList = []
+
+    // Loop over all claims.
+    try {
+      const now = (await web3.eth.getBlock('latest')).timestamp
+      for (let claimID = 0; claimID < contractState.nbClaims; claimID++) {
+        if (
+          claims[claimID] !== undefined &&
+          (
+            (claims[claimID].lastStatus == ClaimStatus.Resolved) ||
+            (claims[claimID].lastStatus == ClaimStatus.Created && now - claims[claimID].lastActionTime < contractState.challengeTimeout) ||
+            (claims[claimID].lastStatus == ClaimStatus.Readjustable && now - claims[claimID].lastActionTime < contractState.readjustmentTimeout) ||
+            (claims[claimID].lastStatus == ClaimStatus.Disputed && now - claims[claimID].lastQueryTime < minQueryInterval)
```

Comment:
I don't understand why we can't just filter out Disputed status altogether? Why do we need minQueryInterval?

Reply:
I tried to reduce the data fetching to the minimum. Maybe it's an overkill. Note that the claimData object (and therefore the claim status) is updated only if the if statement evaluates to false. If I remove the minQueryInterval, the status would change at some point and the bot would never realize it happen.